test/integration/test_main.py

```diff
@@ -175,15 +170,13 @@ def test_role_bad_project_dir():
         ensure_removed("new_logfile")
 
 
-@pytest.mark.serial
-def test_role_run_clean(skipif_pre_ansible28):
+def test_role_run_clean(skipif_pre_ansible28, clear_integration_artifacts):
 
     rc = main(['run', '-r', 'benthomasson.hello_role',
                '--hosts', 'localhost',
                '--roles-path', 'test/integration/roles',
                "test/integration"])
     assert rc == 0
-    ensure_removed("test/integration/artifacts")
```

Comment:
Some of these implemented it in try-except, and others didn't. You're never going to get consistency like this.